### ISSUE TITLE
Startup and Loading experience

### ISSUE THREAD

**Issue by humphd** [mozilla/thimble.webmaker.org#547]
Thimble hosts Brackets, and Brackets takes time to load, and have the UI get fully set up.  In addition, we're going to be moving to a world where we want to load projects from a server, which means download time, writing to fs in browser, and then loading that into the editor.

It would be good to figure out how to use the time and space of the Bramble editor while we're starting.  I notice that http://codepen.io/ chooses to show a gallery in that space, while http://jsfiddle.net/ shows a static image and message saying that it's loading.

We're creating a tool for teaching and learning web development.  Do we want to place a "Did you know?" type thing in that spot that shows while we're loading the editor?  Do we want to put a light gray version of the editor that loads fast/statically and gets replaced by the full editor when it's ready?  Do we want to put the File > New and Open > Project or New > From Template logic in that space, and you choose what you want to do?  Do we just want a better throbber?

I'd love to see us not just sit there for N seconds with a blank white area.

Brackets bugs that might inform this work:
* https://github.com/humphd/brackets/issues/332
* https://github.com/humphd/brackets/issues/312

cc @cassiemc, @hannahkane

**Comment by JordanTheriault** [comment]
I think it's important that this would be a full overlay that sits on a high z-index and also blocks the top bar. Some components being clicked before the app is fully loaded redirects the user to a 404 page while others will spit errors.

I like the idea of interesting facts regarding web development. Perhaps some [cool typographic effects like this](http://www.flukeout.com/examples/_twitter_box/) in addition to a message informing the user the app is loading?

**Comment by humphd** [comment]
The neat part of this is that we can have the Thimble part load almost instantly and be interactive right away without waiting on the rest, since it all lives in a separate iframe.  The entire iframe can be `display:none` if we want, and other content be placed there until it's ready to be swapped in.  I don't think we need to hide the entire page, since it's only part of the app that's going to be slow.

**Comment by JordanTheriault** [comment]
While loading, press the language picker. This is part of Thimble. It should give a 404.

**Comment by hannahkane** [comment]
I like the idea of a "Did you know?" thing, but wonder if it would be frustrating if it actually loaded quite quickly, and the user didn't have time to read the full fact.

This seems like something that @xmatthewx might have some interesting ideas about.

**Comment by xmatthewx** [comment]
@humphd points to two good ways to handle this. jsfiddle's would be quick and easy to implement. Longterm, a gallery within the tool would be great, and probably makes more sense for end users than a gallery on Teach. 

Another option: a window with two big buttons: `Open Project` || `Choose Template` 

Fun facts _could_ be nice if they don't vanish and we can actually build a list of truly fun facts ;)

**Comment by hannahkane** [comment]
+1 to hosting the gallery here, rather than on Teach.

@flukeout and I were just talking about giving Thimble a homepage—it could feature the gallery and some context-setting copy. Plus, of course, a way to access the user's own stuff. This would be a big shift for users who are used to landing directly on a project space, but I think it might make sense.

**Comment by flukeout** [comment]
I'm going to close this one, since we have a loading indicator now. We can file a new issue in the future if we want to improve or change things.
